index.js

```diff
@@ -527,33 +562,88 @@ module.exports = {
             (msg) => {
               const groupId = msg.value.content.recps[0]
 
-              const newKey = Buffer.from(msg.value.content.groupKey, 'base64')
-              ssb.box2.addGroupInfo(groupId, { key: newKey }, (err) => {
+              const secret = Buffer.from(msg.value.content.groupKey, 'base64')
+              ssb.box2.addGroupInfo(groupId, { key: secret }, (err) => {
                 // prettier-ignore
-                if (err) return cb(clarify(err, 'Error adding new epoch key that we found'))
+                if (err) return console.error(clarify(err, 'Error adding new epoch key that we found'))
 
                 const newKeyPick = {
-                  key: newKey,
+                  key: secret,
                   scheme: keySchemes.private_group,
                 }
                 // TODO: naively guessing that this is the latest key for now
                 ssb.box2.pickGroupWriteKey(groupId, newKeyPick, (err) => {
                   // prettier-ignore
-                  if (err) return cb(clarify(err, 'Error switching to new epoch key that we found'))
+                  if (err) return console.error(clarify(err, 'Error switching to new epoch key that we found'))
 
                   ssb.db.reindexEncrypted((err) => {
                     // prettier-ignore
-                    if (err) cb(clarify(err, 'Error reindexing after finding new epoch'))
+                    if (err) console.error(clarify(err, 'Error reindexing after finding new epoch'))
                   })
                 })
               })
             },
             (err) => {
               // prettier-ignore
-              if (err) return cb(clarify(err, "Error finding new epochs we've been added to"))
+              if (err) return console.error(clarify(err, "Error finding new epochs we've been added to"))
             }
           )
         )
+
+        // listen for new epochs and update groupInfo as required
```

Comment:
added this new listener for when you discover a new group epoch or are added to a group.
TODO - this needs merging with the listener above it I think, but I didn't have time to do this